### ISSUE TITLE
Support for converters

### ISSUE THREAD

**Issue by m1raynee** [EQUENOS/disnake#21]
### Summary

Add support for converters for slash command options

### What is the feature request for?

disnake.ext.commands

### The Problem

I cannot eassaly use power of converters in slash commands.

### The Ideal Solution

```py
@commands.slash_command(
        description = '...',
        options = [disnake.Option('name', 'description', disnake.OptionType.string, True, converter=CustomConverter)]
    )
async def test(self, interaction, name):
    ...
```
or
```py
@commands.slash_command(
        description = '...',
        options = [disnake.Option('name', 'description', disnake.OptionType.string, True)]
    )
async def test(self, interaction, name: CustomConverter):
    ...
```

### The Current Solution

```py
@commands.slash_command(...)
async def test(self, interaction, name):
    converted = CustomConverter().convert(interaction, name)
```

### Additional Context

Maybe, for custom converters `slash_convert` (or something different to `convert`) will be called instead of `convert`

**Comment by thesadru** [comment]
I'm against the second option. I think we can avoid the horrible abomination that converters created in normal commands. Most of the time they were all well and good however there was a huge problem when you tried to annotate something as a converter class/callable since it wasn't actually the class and you'd get linter errors. 

If you still want to use something similar to annotations then we may plan on implementing dislash's `OptionParam` into disnake if it turns out to be useful.

**Comment by EQUENOS** [comment]
I'm against the first option. `disnake.Option` is a pure wrapper of an API object, it's not supposed to have attributes that are not related to the original structure defined by the API. My plans were to implement converters as annotations tbh

**Comment by thesadru** [comment]
imo if we're gonna be using annotations then we shouldn't be letting users use the old-style shitty callable converters or converter classes. I admit that it wasn't that big of a deal since everyone just created a converter class which inherited from the return type of the convert method however I don't see a reason to force users to create classes just for that. Or if they are going to be implemented I think we should allow both annotations and `OptionParam`-like fields to also help people who might not want to have problems with their linter for no reason.

**Comment by EQUENOS** [comment]
`OptionParam`-based syntax is good, but right now we're talking about the decorator-based syntax. I think that in this case annotations seem to be the only good solution for converters.

**Comment by thesadru** [comment]
Ah I seem to have misunderstood your point. I think that as long as you have an option to use something else there should be no problem with using annotations.

**Comment by thesadru** [comment]
This has been implemented in #29 

Your code would look like this:
```py
@commands.slash_command()
async def test(self, interaction, name: str = Param(desc="description", conv=custom_converter)):
    ...
```
